src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -1263,6 +1287,18 @@ private void PutConnectionInPool()
             // Make sure there's nothing in the write buffer that should have been flushed
             Debug.Assert(_writeOffset == 0);
 
+            try
+            {
+                Debug.Assert(_initialRead == null, $"Expected a previous initial read to already be consumed");
+                _initialRead = _stream.ReadAsync(_readBuffer, 0, _readBuffer.Length);
```

Comment:
Seems like a comment is warranted here -- I know there's one above, but seems like there should be one here too.